src/main/scala/com/hungry/cars/domain/error/UserError.scala

```diff
@@ -0,0 +1,72 @@
+package com.hungry.cars.domain.error
+
+import io.circe.Encoder
+import io.circe.Json
+import io.circe.syntax.EncoderOps
+
+sealed trait UserError {
+  def code: String
+  def errorMessage: String
+}
+
+object UserError {
+
+  case class UserAlreadyExists(userName: String) extends UserError {
+    def code: String         = "user-error-001"
+    def errorMessage: String = s"Username $userName already exists!"
+  }
+
+  case class UsernameHasSpecialCharacters(userName: String) extends UserError {
+    def code: String         = "user-error-002"
+    def errorMessage: String = s"Username $userName cannot contain special characters."
+  }
+
+  case class EmailIsNotValid(email: String) extends UserError {
+    def code: String         = "user-error-003"
+    def errorMessage: String = "Email is not valid!"
+  }
+
+  case class EmailAlreadyExists(email: String) extends UserError {
+    def code: String         = "user-error-004"
+    def errorMessage: String = s"Email $email already exists!"
+  }
+
+  case object PasswordDoesNotMeetCriteria extends UserError {
+    def code: String = "user-error-005"
+
+    def errorMessage: String =
+      "Password must be at least 8 characters long, including an uppercase, a lowercase letter and one number."
+  }
+
+  case object PasswordsAreNotIdentical extends UserError {
+    def code: String         = "user-error-006"
+    def errorMessage: String = "Passwords are not identical!"
+  }
+
+  case object PasswordCanNotEncrypt extends UserError {
```

Comment:
To nie jest błąd domenowy. Co uzytkownik Twojego API ma z tym zrobić? Ten case sygnalizuje blad programistyczny.